pandas/core/config_init.py

```diff
@@ -460,6 +462,57 @@ def use_inf_as_na_cb(key):
 # Plotting
 # ---------
 
+plotting_backend_doc = """
+: str
+    The plotting backend to use. The default value is "matplotlib", the
+    backend provided with pandas. Other backends can be specified by
+    prodiving the name of the module that implements the backend.
+"""
+
+
+def register_plotting_backend_cb(key):
+    backend_str = cf.get_option(key)
+    if backend_str == 'matplotlib':
+        try:
+            import pandas.plotting._matplotlib  # noqa
+        except ImportError:
+            raise ImportError('matplotlib is required for plotting when the '
+                              'default backend "matplotlib" is selected.')
+        else:
+            return
+
+    try:
```

Comment:
yeah I would just try to import it full-stop; if the backend doesn't support a specific plot then it can rely on our ABC and/or we show a not implemented error. But this regisration whould be a simple import hook.